compiler/passes/normalize.cpp

```diff
@@ -1751,14 +1751,15 @@ static void change_method_into_constructor(FnSymbol* fn) {
 
   // Now check that the function name matches the name of the type
   // attached to 'this'.
```

Comment:
I can't see a difference between if (stringEquals(x, y) == true) ...  and if (strcmp(x, y) == 0) but perhaps I've been programming in C for too long.  The latter is certainly more idiomatic.  The obvious alternative is x == y where x and y are std::string.

Interesting observation about the possible use of astr().
So this could be if (fn->getFormal(2)->type->symbol->name != fn->name && <blah> != astr("init")).
It's more than theoretical that this should work.  It's a bug if it does not. It's tempting to start doing this
more and then shake out any lingering bugs.  This would be easier to rationalize if we didn't declare
the fields to be "const char*" and instead introduced something like Identifier to solidify the intended abstraction.